src/main/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitChecker.java

```diff
@@ -107,7 +108,25 @@ public void checkApiRateLimit(@NonNull TaskListener listener, GitHub github) thr
                 waitUntilRateLimit(listener, github, rateLimit, expiration);
             }
         }
-    };
+    },
+    /**
+     * Ignore GitHub API Rate limit. Useful for GitHub Enterprise instances that might not have a limit set up.
+     */
+    NoThrottle(Messages.ApiRateLimitChecker_NoThrottle()) {
+
+                @Override
+                public void checkApiRateLimit(@NonNull TaskListener listener, GitHub github) throws IOException, InterruptedException {
+                    if (GitHubServerConfig.GITHUB_URL.equals(github.getApiUrl())) {
```

Comment:
Sorry to comment on old PR, for some reason we always hit this If branch for our enterprise github servers, 
any idea why that could happen? is it because we don't have `github.com` defined in our jenkins?